### PR TITLE
#3935 - SIMS to SFAS Bridge - Updates

### PR DESCRIPTION
**Pull request by bidyashish**
**Acceptance Criteria**
- [x] In Application Records: Update Unique Application Key to be Unique Application Number (same character spacing)
- [x] Maintain the same cadence of sending delta changes for these records (including a change to the application id, which could signal a non-assessment impacting change)

Demo: 
Filename: `SIMS-TO-SFAS-20241211-003441.TXT`
Application Number: `2024000046`
image

E2E:
image

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts [hunk 1]
```diff
@@ -138,6 +138,7 @@ export class SIMSToSFASService {
       this.applicationRepo
         .createQueryBuilder("application")
         .select("application.id", "applicationId")
+        .addSelect("application.applicationNumber", "applicationNumber")
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts:141]: I believe we can remove the application.Id as it is not used, in line 140.

File: sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.model.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export type StudentDetail = Student & {
 export interface ApplicationRecord {
   studentId: number;
   applicationId: number;
+  applicationNumber: string;
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.model.ts:11]: Please remove applicationId in line 10

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sims-to-sfas-integration.scheduler.e2e-spec.ts [hunk 3]
```diff
@@ -798,11 +798,9 @@ describe(describeProcessorRootTest(QueueNames.SIMSToSFASIntegration), () => {
       application.applicationStatus === ApplicationStatus.Cancelled
         ? formatDate(application.applicationStatusUpdatedOn, DATE_FORMAT)
         : "        ";
-    return `${applicationRecordType}${studentId
-      .toString()
-      .padStart(10, "0")}${application.id
-      .toString()
-      .padStart(10, "0")}${formatDate(studyStartDate, DATE_FORMAT)}${formatDate(
+    return `${applicationRecordType}${studentId.toString().padStart(10, "0")}${
+      application.applicationNumber
```
Comment (guru-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sims-to-sfas-integration.scheduler.e2e-spec.ts:802]: 👍

**Review by dheepak-aot — APPROVED** [review]
Good Job @bidyashish Please take care of the comments mentioned by @guru-aot

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @bidyashish

**Pull request merged**